### PR TITLE
Object namespace for overriding the namespace in the helm chart added

### LINKED ISSUE #20511: CFP: Add a value for overriding the namespace in the helm chart.
## Cilium Feature Proposal
        
Thanks for taking time to make a feature request for Cilium! If you have usage questions, please try the [slack channel](http://slack.cilium.io/) and see the [FAQ](https://github.com/cilium/cilium/issues?utf8=%E2%9C%93&q=is:issue+label:kind/question+) first.

**Is your feature request related to a problem?**

Adding cilium to an umbrella helm chart but installing to a different namespace isn't currently possible.

**Describe the feature you'd like**

Adding `namespaceOverride: ""` to values.yaml (similar to how bitnami makes their helm charts) or how cert-manager adds the 'namespace' value.

bitnami example:
```
## @param namespaceOverride String to fully override common.names.namespace
##
namespaceOverride: ""
```

cert-manager example: 
```
# This namespace allows you to define where the services will be installed into
# if not set then they will use the namespace of the release
# This is helpful when installing cert manager as a chart dependency (sub chart)
namespace: ""
```

**(Optional) Describe your proposed solution**

I think either the way bitnami or cert-manager handles it works well and could be useful.

### PR DESCRIPTION
**Pull request by basit9958**
Signed-off-by: Basit Hasan <hasanbasit13@gmail.com>

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

 Added namespace object to let helm support installation of services in other namespaces as well.  

Fixes: #20511

### CODE REVIEW

**Review by gandro — CHANGES_REQUESTED** [review]
Thanks for the PR. This however does not solve the linked issue, since the value is not used by any component.

You will have to introduce a Helm helper similar to `common.names.namespace` the Bitnami charts and more importantly use it for the `namespace:` property of every Kubernetes resource we deploy in our charts. I would propose `cilium.namespace` as the name of the helper.

https://github.com/bitnami/charts/blob/ab3db922c645f313f547a227b77f2f3b06c4a7ed/bitnami/common/templates/_names.tpl#L54-L63

I personally also prefer `.Values.namespaceOverride` over `.Values.namespace`, but that's up for debate.

File: install/kubernetes/cilium/templates/cilium-agent/clusterrolebinding.yaml [hunk 1]
```diff
@@ -10,5 +10,5 @@ roleRef:
 subjects:
 - kind: ServiceAccount
   name: {{ .Values.serviceAccounts.cilium.name | quote }}
-  namespace: {{ .Release.Namespace }}
+  namespace: {{ cilium.namespace }}
```
Comment (gandro) [inline install/kubernetes/cilium/templates/cilium-agent/clusterrolebinding.yaml:13]: This looks incorrect unfortunately. To use the defined template, you'll have to call include: https://helm.sh/docs/chart_template_guide/named_templates/#the-include-function

**Review by gandro — CHANGES_REQUESTED** [review]
Thanks for addressing the above feedback. Still some more changes needed.

**Comment by pchaigno** [comment]
Switching to draft while the review is being addressed.

**Pull request closed without merging**